dask/array/tests/test_array_core.py

```diff
@@ -1611,6 +1611,16 @@ def test_slice_with_floats():
         d[[1, 1.5]]
 
 
+def test_slice_with_uint():
+    d = da.ones((5,), chunks=(3,))
+    assert_eq(d[np.array([0])],
+              d[np.array([0], dtype='uint64')])
+    assert_eq(d[np.array([0])],
+              d[np.array([0], dtype='uint32')])
+    assert_eq(d[np.array([0])],
+              d[np.array([0], dtype='uint16')])
```

Comment:
Since `d` is all ones, any index would have the same result. I also would test against numpy instead of dask (and don't think we need to test all the unsigned dtypes). Perhaps:

```python
def test_slice_with_uint():
    x = np.arange(10)
    dx = da.from_array(x, chunks=5)
    inds = np.array([0, 3, 6], dtype='u8')
    assert_eq(dx[inds], x[inds])
    assert_eq(dx[inds.astype('u4')], x[inds.astype('u4')])
```